test/ccapi/unittest_ccapi.cpp

```diff
@@ -371,9 +371,9 @@ TEST(nntrainer_ccapi, train_batch_size_update_after) {
   EXPECT_EQ(model->setProperty({"batch_size=4"}), ML_ERROR_NONE);
   EXPECT_NO_THROW(model->train());
 
-  EXPECT_FLOAT_EQ(model->getTrainingLoss(), 1.9582682);
-  EXPECT_FLOAT_EQ(model->getValidationLoss(), 2.1831701);
-  EXPECT_FLOAT_EQ(model->getLoss(), 2.1985414);
+  EXPECT_FLOAT_EQ(model->getTrainingLoss(), 1.9613363);
```

Comment:
As far as I can tell, initialize doesn't have reinitialize, may be because the init order changed so affect the loss?

Reply:
The issue is not from the initialize but the optimizer.
In the older version, the memory was allocated again and again, which initialized the adam parameters from the scratch for each run.
With this patch, the memory is not re-allocated. And the adam optimizer parameters start from the values of the previous iteration.

Reply:
I see, thanks for the explanation :)